private/lib/bzlmod.bzl

```diff
@@ -0,0 +1,32 @@
+# Utility functions for handling bzlmod-related features
+
+def to_visible_name(name):
+    """Convert a name (typically from `Label.workspace_name`) to the human-readable version.
+
+    `bzlmod` will mangle the names of repos, but it's often useful to have the human-readable
+    version of the name available.
+
+    Args:
+      name: (string) The name to convert, often this will be from `Label.workspace_name`
+    """
+
+    if "+" in name:
+        return name.rpartition("+")[-1]
+
+    if "~" in name:
+        return name.rpartition("~")[-1]
+
+    return name
+
+def get_file_owner_repo_name(name):
+    """Convert a name (typically from `File.owner.workspace_name`) to the owning repo
```

Comment:
This doesn't return the name of a repo, it gives you the name of the module that owns the repo that owns the file.

Note that this can have surprising effects: If my `better_repo_rules` module offers a `better_http_archive` module extension, then the reported owner of files in such archives will always be `better_repo_rules`, not the module that specified tags for the extension.

@Wyverald What do you think of exposing a `module_name` field on `Label`? That wouldn't require any repo unmapping, which is nice.

Reply:
I've updated the method name and comment.